test/gen-matter-4.test.js

```diff
@@ -83,6 +83,80 @@ test(
       { disableDeprecationWarnings: true }
     )
     expect(genResult.hasErrors).toEqual(false)
+
+    // Note: A lot of these tests here are for sake of backwards
+    // compatibility. Latest ZAP must be able to generate content for
+    // the old SDKs, so if you changed something that generates
+    // endpoint_config differently, please be very very careful and
+    // make yourself you can answer positively the following question:
+    //   will after my changes, zap still be able to generate content
+    //   that will work with an older SDK.
+    //
+    let ept = genResult.content['endpoint_config.h']
+    expect(ept).toContain(`{ \\
+      /* Endpoint: 65534, Cluster: Network Commissioning (server) */ \\
```

Comment:
Shouldn't we also have some tests for non-default `order` values?  Here and in other places.

Reply:
I'll add some tests for that `order="...."` into the template.